include/zephyr/dt-bindings/inputmux/nxp_inputmux_rt5xx.h

```diff
@@ -0,0 +1,907 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __DT_BINDINGS_INPUTMUX_NXP_INPUTMUX_RT5XX_H__
+#define __DT_BINDINGS_INPUTMUX_NXP_INPUTMUX_RT5XX_H__
+
+#define SCT0_PMUX_ID         0x00U
+#define PINTSEL_PMUX_ID      0x100U
+#define DSP_INT_PMUX_ID      0x140U
+#define DMA0_ITRIG_PMUX_ID   0x200U
+#define DMA0_OTRIG_PMUX_ID   0x300U
+#define DMA0_CHMUX_SEL0_ID   0x320U
+#define DMA1_ITRIG_PMUX_ID   0x400U
+#define DMA1_OTRIG_PMUX_ID   0x500U
+#define DMA1_CHMUX_SEL0_ID   0x520U
+#define CT32BIT0_CAP_PMUX_ID 0x600U
+#define CT32BIT1_CAP_PMUX_ID 0x610U
+#define CT32BIT2_CAP_PMUX_ID 0x620U
+#define CT32BIT3_CAP_PMUX_ID 0x630U
+#define CT32BIT4_CAP_PMUX_ID 0x640U
+#define FREQMEAS_PMUX_ID     0x700U
+#define SMART_DMA_PMUX_ID    0x720U
+#define DMA0_REQ_ENA0_ID     0x740U
+#define DMA1_REQ_ENA0_ID     0x760U
+#define DMA0_ITRIG_EN0_ID    0x780U
+#define DMA1_ITRIG_EN0_ID    0x7A0U
+#define ENA_SHIFT       5U
+#define PMUX_SHIFT      20U
+#define CHMUX_AVL_SHIFT 31U
+#define CHMUX_OFF_SHIFT 19U
+#define CHMUX_VAL_SHIFT 17U
+
+/* Definitions for SCT0 */
+#define MUX_SCT0_PIN_INP0_TO_SCT0                    (0U + (SCT0_PMUX_ID << PMUX_SHIFT))
+#define MUX_SCT0_PIN_INP1_TO_SCT0                    (1U + (SCT0_PMUX_ID << PMUX_SHIFT))
+#define MUX_SCT0_PIN_INP2_TO_SCT0                    (2U + (SCT0_PMUX_ID << PMUX_SHIFT))
+#define MUX_SCT0_PIN_INP3_TO_SCT0                    (3U + (SCT0_PMUX_ID << PMUX_SHIFT))
```

Comment:
all of the macros in these files are super inefficient, we can use a few macro functions and eliminate 90% of the lines in the files and make it less confusing for users

Reply:
These files are being generated from the `fsl_inputmux_connections.h` file within the HAL. I agree they are verbose, but I think manually creating these files is going to be a pain and error prone.